lib/python/qmk/decorators.py

```diff
@@ -82,3 +83,34 @@ def wrapped_func(*args, **kwargs):
         return wrapped_func
 
     return wrapper_cache
+
+
+def profile_code(func):
+    """Profile the execution of a function.
+
+    This will print out a call graph of the function's execution.
+    """
+    @contextlib.contextmanager
+    def profile_execution():
+        try:
+            profiling = False
+            try:
+                import yappi
+                yappi.start()
+                profiling = True
+            except ImportError:
+                pass
```

Comment:
Idiomatic and shorter
```suggestion
        with contextlib.suppress(ImportError):
            import yappi
            yappi.start()
            profiling = True
```
Written on phone, forgive potential typos